includes/admin/settings/SameDayDeliveryService.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\WooCommerce\includes\admin\settings;
+
+use DateTime;
+use DateTimeZone;
+use WCMYPA_Settings;
+
+/**
+ * Service to determine whether the sameDayDelivery option should be displayed in the delivery options widget.
+ */
+class SameDayDeliveryService
+{
+    /**
+     * @var string
+     */
+    private $carrierName;
+
+    /**
+     * @var
+     */
+    private $settingsCollection;
+
+    /**
+     * @param  string $carrierName
+     */
+    public function __construct(string $carrierName)
+    {
+        $this->carrierName        = $carrierName;
+        $this->settingsCollection = WCMYPA()->setting_collection->where('carrier', $carrierName);
+    }
+
+    /**
+     * @return bool
+     */
+    public function shouldShowSameDayDelivery(): bool
+    {
+        $settingCollection    = WCMYPA()->setting_collection->where('carrier', $this->carrierName);
+        $carrierIsActive      = (bool) $settingCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DELIVERY_ENABLED);
+        $sameDayFromSettings  = (bool) $settingCollection->getByName(WCMYPA_Settings::SETTING_CARRIER_DEFAULT_EXPORT_SAME_DAY_DELIVERY);
+
+        if (! $carrierIsActive && ! $sameDayFromSettings) {
```

Comment:
Ik weet bijna zeker dat dit || moet zijn ipv &&, graag even dubbelchecken? Waarom zou je anders nog de same day setting willen controleren, als je carrier niet active is...